### DISCUSSION TITLE
Campaign difficulty selection

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#5666] [Ideas]
There were some rumors, that in future a campaign difficulty adjustment is going to be implemented.
I want to rise here two questions about how this is going to happen.
First of all we should decide whether we allow to select difficulty for the whole campaign right before we start or change difficulty level individually for each scenario.
Secondly, what gameplay aspects should affect the selected difficulty level.

**Comment by Branikolog** [comment]
I believe, that selecting game difficulty should happen only once per game before starting the campaign. The majority of games doesn't allow to change difficulty during game process. Also, playing on a static difficulty (selected in the beginning of the campaign walkthrough) would make consistent game perception and would offer players a challenge of finishing a campaign on a particular difficulty level, higher, than the previous attempt. Instead of mess with randomly adjusted scenarios. This also can make a challenge for a player to finish campaign on all difficulties. Otherwise player couldn't remember exact difficulty level he selected on a particular scenario during previous attempt.
Farther, final highscore would be another problem, when game allows a player to change difficulty during walkthrough. This would lead to depreciating of highscores value. While contrariwise we can make a separate highscore list for various difficulties or make some kind of mark, showing that player finished the campaign on a particular "extremely insane high" difficulty. :)
As for "cons" of my idea:  if a player face some real troubles with our advanced AI he will have to start a campaign from the beginning on more suitable difficulty level. :)

Talking about difficulty adjustment itself.
Difficulty level in the OG affected only starting resources and AI "cheats". :) Also, it's a well known fact, that each campaign scenario has its own difficulty level, made by the original developers which couldn't be adjusted.
In my personal opinion, we should respect the in-game difficulty set by original developers (regarding starting resources) and raise difficulty only by increasing AI algorithms and adjusting some implicit mechanics, but not affecting the amount of resources player receives on scenario start. It should work like difficulty of single scenario: where we can select game difficulty level (including starting resources etc.) and also we can watch fixed difficulty level of particular map set by map creator himself (which seems to me close to the original fixed campaign difficulty level). So I think original developers offered us a particular fixed difficulty level (regarding starting resources) which allows the map to be passed. So fheroes2 difficulty adjustment should affect only the AI and some other gameplay twirks to make walkthrough more tough. Changing the amount of starting resources would make some scenarios too boring (in a case player chooses "easy" or "normal", while the particular scenario was marked as "expert" in the original game. Or, on contrary, make some scenarios impossible without starting resources, since the original developers often tend to grant AI lots of army and resources trying to make opponents threat at least a little closer to human player. 
In a case, the scenario becomes impossible, player will be forced to change difficulty level of a few particular scenarios which will look like bad, unplayable map design. So, in a case difficulty level was set in the beginning of the campaign - player would have to start from the beginning, which is not good. To prevent this, we can manually change something on a map or adjust gameplay twirks for a particular scenario, but it would be quite inefficient solution of the problem. Better is to improve the difficulty for all maps together, rather then adjust it for each unplayable map later.
I think we shouldn't change starting resources as it could drastically affect playability of some scenarios.
In my personal opinion, we shouldn't make original campaigns insanely difficult and unplayable on higher difficulty levels. It would be enough, if an experienced player will complete campaign with a few restarts or something like that.

**Comment by oleg-derevenetz** [comment]
It's good that I never listen to rumors :) Life becomes calmer.

**Comment by Branikolog** [comment]
Closing as difficulty selection was implemented long time ago.
